server/routes/index.js

```diff
@@ -0,0 +1,106 @@
+const fs = require('fs')
+
+const each = require('async/each')
+const pick = require('lodash.pick')
+
+const routes = {}
+module.exports = routes
+
+routes.makeRoutes = (app) => {
+  fs.readdirSync(__dirname).forEach((file) => {
+    if (file === 'index.js') return
+    const name = file.substr(0, file.indexOf('.'))
+    require('./' + name)(app)
+  })
+}
+
+const serverError = (res, err) => {
+  console.error(err)
+  res.status(500).json({error: err})
+}
+
+const makeGenericModelResponse = (res) => (err, data) => {
+  res.set('Content-Type', 'application/json')
+  if (err) return serverError(res, err)
+  res.json(data)
+}
+
+const makeGetModelResponse = (childModels, res, isCollection) => (err, data) => {
+  const genericResponder = makeGenericModelResponse(res)
+  if (!Array.isArray(data)) {
+    data = [data]
+  }
+
+  if (!childModels) return genericResponder(err, isCollection ? data : data[0])
+
+  // wow, awesome, with normalized mongoose models I get to make extra queries
+  // to the db to do joins!  </sarcasm>
```

Comment:
These less than 20 lines of code must have been very painful for you to write :)

Also, you probably want to use [populate](http://mongoosejs.com/docs/populate.html)